src/Contao/SessionStorage.php

```diff
@@ -44,26 +48,55 @@ class SessionStorage implements SessionStorageInterface
      */
     private $session;
 
+    /**
+     * The database keys for store session data in the database.
+     *
+     * @var array
+     */
+    private $databaseKeys = [];
+
     /**
      * The attribute storage.
      *
      * @var array
      */
-    private $attributes;
+    private $attributes = [];
 
     /**
      * Create a new instance.
      *
-     * @param string           $key     The key to use for storage.
-     *
-     * @param SessionInterface $session The symfony session.
+     * @param RequestScopeDeterminator $scopeDeterminator The request scope determinator.
+     * @param string                   $key               The key to use for storage.
+     * @param SessionInterface|null    $session           The symfony session.
+     * @param array                    $databaseKeys      The database keys for store session data in the database.
      */
-    public function __construct($key = '', SessionInterface $session = null)
```

Comment:
Why can the `$session` be `null`? Where is the fallback loaded (It was this way before, but it never get's loaded?)

Reply:
I have rework this

Reply:
b54c94d3